x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/shared/FrozenCacheService.java

```diff
@@ -196,7 +256,13 @@ public void validate(final ByteSizeValue value, final Map<Setting<?>, Object> se
     @SuppressWarnings({ "unchecked", "rawtypes" })
     public FrozenCacheService(NodeEnvironment environment, Settings settings, ThreadPool threadPool) {
         this.currentTimeSupplier = threadPool::relativeTimeInMillis;
-        this.cacheSize = SNAPSHOT_CACHE_SIZE_SETTING.get(settings).getBytes();
+        FsInfo.Path pathInfo;
+        try {
+            pathInfo = FsProbe.getFSInfo(environment.nodePaths()[0]);
```

Comment:
`environment.nodeDataPaths()[0]` here

Reply:
Fixed in 7280ae8789d3df43667aae9501815c6b1c7ec3e1